Projects/Social-Trading/SA/Function-Libraries/SocialEntitiesProfile.js

```diff
@@ -473,6 +478,48 @@ exports.newSocialTradingFunctionLibrariesSocialEntitiesProfile = function () {
 
                 SA.projects.network.globals.memory.maps.USER_PROFILES_BY_ID.set(userProfile.id, inMemoryUserProfile)
             }
+            
+            function saveApiAppFile() {
+             
+                const SOCIAL_TRADING_REPO_NAME = profileMessage.socialEntityHandle + "-" + profileMessage.socialEntityType.replace(' ', '-') + "-Data"
+                /*
+                Try to load the Secrets file. 
+                */
+                let filePath = global.env.PATH_TO_SECRETS + '/'
+                let fileName = "ApisSecrets.json"
+                let fileContent 
+                try {
+                    fileContent = SA.nodeModules.fs.readFileSync(filePath + '/' + fileName)
```

Comment:
You should check if the file exists before reading so that your catch block can function as it should.

// Check Secrets Folder Exists and if not create it
if (!fs.existsSync(filePath) {
    SA.projects.foundations.utilities.filesAndDirectories.createNewDir(filePath);
}

// Check if ApiSecrets file exists
if (fs.existsSync(filePath + '/' + fileName) ) {
    fileContent = SA.nodeModules.fs.readFileSync(filePath + '/' + fileName)
} else {
    SA.logger.warn('ApisSecrets file has not been created yet.');
    fileContent = undefined;
}


if (fileContent !== undefined) {
       secretsFile = JSON.parse(fileContent)
} else {
    secretsFile = {
             secrets: []
            }
}

Reply:
The checks that @theblockchainarborist suggested aren't there yet. @end-debug Basically the steps that need to happen before and after loading are: 

1. Check if the file exists. If it doesn't create the file 

2.

Reply:
check out the commit I made with the checks from your list